### PR TITLE
Adding magicgui recipe

### PR DESCRIPTION
**Pull request by k-dominik**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by k-dominik** [comment]
Hi @conda-forge/help-python,

this is ready to be reviewed! :)

File: recipes/magicgui/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install --no-deps --ignore-installed . -vv"
+
+requirements:
+  host:
+    - pip
+    - python
+    - setuptools_scm
+    - qtpy >=1.7.0
```
Comment (scopatz) [inline recipes/magicgui/meta.yaml:22]: Do you really need qtpy at build time?

Comment (k-dominik) [reply recipes/magicgui/meta.yaml]: Hi @scopatz,

thank you very much for spotting this. I don't know how it ended up there :)

File: recipes/magicgui/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  license_family: MIT
+  license_file: LICENSE
+  summary: "build GUIs from functions, using magic."
+  doc_url: https://magicgui.readthedocs.io/en/latest/
+  dev_url: https://github.com/napari/magicgui
+
+extra:
+  recipe-maintainers:
+    - k-dominik
+    - tlambert03
```
Comment (scopatz) [inline recipes/magicgui/meta.yaml:44]: @tlambert03 - are you OK with being a maintainer?

Comment (tlambert03) [reply recipes/magicgui/meta.yaml:44]: Yes, I wrote the package.  Thanks for checking!

**Comment by scopatz** [comment]
Thanks!

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**